### ISSUE TITLE
Broken code involing predicates and number comparison

### ISSUE THREAD

**Issue by stepancheg** [jckarter/clay#482]
This code is broken by commit 09fabf90742c720cadff77294092a3f33fbca277.

```
[..F when (countValues(..F) > 0)]
referenceType(..fields:F) = recordWithProperties(
    [],
    ["ptr", Int]
);


record ReferenceRecord = referenceType(x: Int);
overload equals?(a:ReferenceRecord, b:ReferenceRecord) = a.x == b.x;

main() {
}
```

Error is:

```
###############################


[A,B when not equals?(A,B) and Numeric?(A,B)]
---------------------^
forceinline overload (>)(a:A, b:B) : Bool =
    (>)(toBiggerNumericType(A, B, a), toBiggerNumericType(A, B, b));
###############################
/Users/yozh/devel/left/clay/lib-clay/core/numbers/numbers.clay(245,21): error: predicate evaluation loop

compilation context: 
    ==(Static[Int32], Static[Int32])
  /Users/yozh/devel/left/clay/lib-clay/core/numbers/numbers.clay(245,21):
    >(Int32, Int32)
  /Users/yozh/devel/left/clay/lib-clay/core/operators/operators.clay(429,18):
    infixOperator(Int32, Static[>], Int32)
  tmp.clay(1,28):
    referenceType(Tuple[Static["x"], Static[Int32]])
  tmp.clay(8,38):
    ReferenceRecord
  tmp.clay(9,19):
    ==(Static[Int32], Static[Int32])
  /Users/yozh/devel/left/clay/lib-clay/core/numbers/numbers.clay(253,22):
    >=(Int32, Int32)
  /Users/yozh/devel/left/clay/lib-clay/core/operators/operators.clay(429,18):
    infixOperator(Int32, Static[>=], Int32)
  /Users/yozh/devel/left/clay/lib-clay/core/pointers/pointers.clay(246,45):
    call(ExternalCodePointer[AttributeCCall, true, [Pointer[Struct___sFILE], Pointer[Int8]], [Int32]], Pointer[Struct___sFILE], Pointer[Int8])
  /Users/yozh/devel/left/clay/lib-clay/core/exceptions/exceptions.clay(156,16):
    errorMsg(Static["exception when initializing globals\n"])
```

**Comment by ghost** [comment]
Should be fixed with commit f1782171fff2bc13a05f81ea8a69e69ff93225a9.

**Comment by stepancheg** [comment]
@agemogolk Thanks.
